### ISSUE TITLE
Unable to connect to Azure's Cosmos DB (MongoDB)

### ISSUE THREAD

**Issue by techknowlogick** [cesanta/docker_auth#205]
Azure runs Cosmos DB on a non-standard MongoDB port, and requires SSL. Due to these two things docker_auth is unable to connect to the MongoDB server.

Example of the config block that I attempted to use to connect so far:
```
mongo_auth:
  dial_info:
    addrs: ["docker-registry-auth-test.documents.azure.com:10255"]
    timeout: "10s"
    database: "docker_auth"
    username: "docker-registry-auth-test"
    password: "abcd1234"
  collection: "users"
```
Azure provides a connection string similar to: `mongodb://docker-registry-auth-test:abcd1234@docker-registry-auth-test.documents.azure.com:10255/?ssl=true`

**Comment by rojer** [comment]
@carsonoid any ideas?

**Comment by techknowlogick** [comment]
Ah, a dial server function needs to be added to the dialfunction.

Here is some sample code that Microsoft provides: https://docs.microsoft.com/en-us/azure/cosmos-db/create-mongodb-golang#review-the-code

perhaps a bool under dial_info could be added (just like how password_file is non-standard) for tls, then if that is set to true use a tlsdialer.

**Comment by carsonoid** [comment]
Yeah that would be easy enough to add. I can probably write up a PR for
that if you aren't able/willing to do it.

**Comment by techknowlogick** [comment]
I've just created a PR for this. Please let me know if I need to sign a CLA or something.

**Comment by rojer** [comment]
i built and pushed a new image to `:latest`, you can try it there.

**Comment by techknowlogick** [comment]
It worked! Thank you :heart:

Just a note for people googling this: I wasn't able to use the read-only keys from Azure Cosmos DB, as the following error happened: `Cannot execute command createindexes using PrimaryReadonlyMasterKey`, as well once I changed to the read-write key it also error'd out because: `{"Errors":["Unique Index can be modified only if collection is empty"]}`. However, I removed everything from my collection (it only had one record at the time), ran the auth server, the indices were created, and then put back in the record (per standard mongodb format), and I was able to login.

TLDR: Use read-write keys, and make sure the collection is empty before you start the auth_server.
